### ISSUE TITLE
Add a fontSizeThatFits() method!

### ISSUE THREAD

**Issue by Sweeper777** [tbaranes/FittableFontLabel#12]
The label that I'm using is a little special. I can't just set the text of it and call `fontSizeToFit` to get the font size that fits. Can you add a method in the `UILabel` extension that returns a font size of the argument text that fits the label's size.

The signature is like this:

```
public func fontSizeThatFits(text: String, size: CGSize = self.bounds.size) -> CGFloat
```

The basic idea is that I give you some text, and the method will figure out how big the text should be to just fit the size of the label.

I think you must have this as a private method, right? Can you just make it public?

**Comment by tbaranes** [comment]
I won't have time soon to do it, but I'm very welcome to review a PR 👍 
It's a good idea, and shouldn't be too complex to implement.

In the current implementation, you can refactor the code to externalise that part, but the method doesn't exist _directly_

**Comment by Sweeper777** [comment]
Okay, this is the method that I wrote. I mostly just copied the logic from the private `fontSizeToFit` method.

```
public func fontSizeThatFits(text string: String, maxFontSize maxFontSize: CGFloat = CGFloat.NaN, minFontScale: CGFloat = 0.1,rectSize: CGSize? = nil) -> CGFloat {
    let maxFontSize = maxFontSize.isNaN ? 100 : maxFontSize
    let minFontScale = minFontScale.isNaN ? 0.1 : minFontScale
    let rectSize = rectSize ?? bounds.size

    var newAttributes = currentAttributedStringAttributes()
    guard string.characters.count != 0 && newAttributes.count > 0 else {
        return self.font.pointSize
    }

    let minimumFontSize = maxFontSize * minFontScale
    let boundingSize = numberOfLines == 1 ? CGSize(width: CGFloat.max, height: rectSize.height) : CGSize(width: rectSize.width, height: CGFloat.max)
    var newFont = UIFont()
    var fontSize = maxFontSize
    repeat {
        newFont = font.fontWithSize(fontSize)
        newAttributes[NSFontAttributeName] = newFont
        let area = string.boundingRectWithSize(boundingSize, options: .UsesLineFragmentOrigin, attributes: newAttributes, context: nil).size
        if (numberOfLines == 1 && area.width <= rectSize.width) ||
            (numberOfLines != 1 && area.height <= rectSize.height) {
            break
        }

        fontSize -= 1
        if fontSize < minimumFontSize {
            fontSize = minimumFontSize
        }
    } while fontSize > minimumFontSize
    return fontSize
}
```

I tested it with a long string and a short string. And as expected, the long string returns a smaller size and the short one returns a larger size.

That's all the testing I've done so far. I can't think of any other way to test it. Can you spot any flaws or pitfalls?

**Comment by tbaranes** [comment]
Seems good 👍 
Another way to test it would be to make the current `fontSizeToFit` use that new method and see if anything is broken in the demo app. Otherwise, for the basics, testing a long and a small string seems enough to me.

**Comment by Sweeper777** [comment]
Pull request created: #13

**Comment by tbaranes** [comment]
Just released 1.2.0 which including your feature! Thanks again 👍
